streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskEOSIntegrationTest.java

```diff
@@ -82,29 +82,27 @@ public void surviveWithOneTaskAsStandby() throws ExecutionException, Interrupted
 
         final CountDownLatch instanceLatch = new CountDownLatch(1);
 
-        final String stateDirPathOne = stateDirPath + "/" + appId + "-1/";
-        final KafkaStreams streamInstanceOne =
-            buildStreamWithDirtyStateDir(appId, stateDirPathOne, instanceLatch);
+        try (
+            final KafkaStreams streamInstanceOne = buildStreamWithDirtyStateDir(appId, stateDirPath + "/" + appId + "-1/", instanceLatch);
+            final KafkaStreams streamInstanceTwo = buildStreamWithDirtyStateDir(appId, stateDirPath + "/" + appId + "-2/", instanceLatch);
+        ) {
 
-        final String stateDirPathTwo = stateDirPath + "/" + appId + "-2/";
-        final KafkaStreams streamInstanceTwo =
-            buildStreamWithDirtyStateDir(appId, stateDirPathTwo, instanceLatch);
 
-        streamInstanceOne.start();
+            streamInstanceOne.start();
 
-        streamInstanceTwo.start();
+            streamInstanceTwo.start();
 
-        // Wait for the record to be processed
-        assertTrue(instanceLatch.await(15, TimeUnit.SECONDS));
+            // Wait for the record to be processed
+            assertTrue(instanceLatch.await(15, TimeUnit.SECONDS));
 
-        waitForCondition(() -> streamInstanceOne.state().equals(KafkaStreams.State.RUNNING),
-            "Stream instance one should be up and running by now");
-        waitForCondition(() -> streamInstanceTwo.state().equals(KafkaStreams.State.RUNNING),
-            "Stream instance one should be up and running by now");
-
-        streamInstanceOne.close(Duration.ofSeconds(30));
-        streamInstanceTwo.close(Duration.ofSeconds(30));
+            waitForCondition(() -> streamInstanceOne.state().equals(KafkaStreams.State.RUNNING),
+                             "Stream instance one should be up and running by now");
+            waitForCondition(() -> streamInstanceTwo.state().equals(KafkaStreams.State.RUNNING),
+                             "Stream instance one should be up and running by now");
 
+            streamInstanceOne.close(Duration.ZERO);
+            streamInstanceTwo.close(Duration.ZERO);
```

Comment:
Close them both asynchronously so we can tear them down in parallel. The try-with-resources will block until they are really closed.